### ISSUE TITLE
Fatal error: 

### ISSUE THREAD

**Issue by ginblue** [olefredrik/FoundationPress#308]
Have just started developing a theme with Foundationpress. Nothing has been done other than set up a static 'front-page.php' and change text domain. The following error is now being thrown on front page and post pages. Seems to have been caused by the front page somehow.

Fatal error: Can't use function return value in write context in C:\Program Files\Ampps\www\retro-resources-v4\wp-content\themes\retroresouces\comments.php on line 39

If anyone can shed some light on this issue it would be much appreciated.

Anthony

**Comment by olefredrik** [comment]
@josh-rathke : any ideas? 

Referring to issue https://github.com/olefredrik/FoundationPress/issues/297 is seems like this bug was introduced in this commit https://github.com/olefredrik/FoundationPress/commit/7acc16c27c7420e08d70c777740e1ee7690cd282

**Comment by ginblue** [comment]
Have adopted @LukePettway solution, which seems to work for me.

**Comment by Aetles** [comment]
I wonder if `sanitize_file_name()` is necessary at all in that code? 

We are only checking if the file name is exactly `comments.php` so I'm not sure what kind of added security a sanitizing of `$_SERVER['SCRIPT_FILENAME']` would bring in this case, but maybe @josh-rathke can explain it?

**Comment by josh-rathke** [comment]
The reason I added the sanitization functions is because without them, the build does not pass WordPress Coding Standards. This is the error that it throws up:

'''FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 39 | ERROR | Detected usage of a non-sanitized input variable:
    |       | $_SERVER (WordPress.VIP.ValidatedSanitizedInput.)
----------------------------------------------------------------------'''

Further research on that particular error code will eventually bring you to [this](https://vip.wordpress.com/documentation/best-practices/security/validating-sanitizing-escaping/) site which explains the how and why, but it would seem that it's geared far more towards user input:

**Comment by Aetles** [comment]
Ah, yeah it's definitely useful for files uploaded from users, "Never trust user input" is a good mantra. :) In this case it seemed unnecessary, but if it does not pass the check then maybe [the solution](https://github.com/olefredrik/FoundationPress/issues/297#issuecomment-92897116) by LukePettway is the way to go? 

Or maybe there's a better way than checking `$_SERVER['SCRIPT_FILENAME']`? The WP Codex for writing a plugin [suggests another way](https://codex.wordpress.org/Writing_a_Plugin#Plugin_Files) to prevent direct access:

> **Security Note**: Consider blocking direct access to your plugin PHP files by adding the following line at the top of each of them, or be sure to refrain from executing sensitive standalone PHP code before calling any WordPress functions.

> `defined( 'ABSPATH' ) or die( 'No script kiddies please!' );`


I don't know if there's any downside to it, though, but it seems to be [suggested](http://stackoverflow.com/questions/27250343/security-wordpress-template) for templates as well.

And btw, line 40:
```php
		die (__( 'Please do not load this page directly. Thanks!', 'FoundationPress' )); }
```
should be
```php
			die ( __( 'Please do not load this page directly. Thanks!', 'FoundationPress' ) ); 
		}
```

;)

**Comment by josh-rathke** [comment]
I think using the WP Codex's method is cleaner. It makes more sense on an
initial read through as well.

**Comment by josh-rathke** [comment]
Also Fixes #297
